src/components/Dropdown.tsx

```diff
@@ -0,0 +1,107 @@
+import {
+  FC,
+  useState,
+  ChangeEvent,
+  FocusEvent,
+  memo,
+  useCallback,
+} from 'react';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+
+import { Person } from '../types/Person';
+
+import './Dropdown.scss';
+
+interface Props {
+  people: Person[];
+  delay?: number;
+  onChangeQuery: (query: string) => void;
+  onSelectPerson: (person: Person | null) => void;
+}
+
+const Dropdown: FC<Props> = ({
+  people,
+  delay = 300,
+  onSelectPerson,
+  onChangeQuery,
+}) => {
+  const [openDropdown, setOpenDropdown] = useState(false);
+  const [searchValue, setSearchValue] = useState('');
+
+  const debouncedOnChangeQuery = useCallback(
+    debounce((str: string) => {
+      onChangeQuery(str);
+    }, delay),
+    [delay, onChangeQuery],
+  );
+
+  const handleQueryChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setSearchValue(event.target.value);
+    onSelectPerson(null);
+    debouncedOnChangeQuery(event.target.value);
+  };
+
+  const handleFocus = () => {
+    setOpenDropdown(prev => !prev);
+  };
+
+  const handleBlur = (event: FocusEvent<HTMLInputElement, Element>) => {
+    if (!event.relatedTarget) {
+      setOpenDropdown(prev => !prev);
+    }
+  };
+
+  const handleClick = (person: Person) => {
+    setSearchValue(person.name);
+    onSelectPerson(person);
+    onChangeQuery('');
+    setOpenDropdown(prev => !prev);
```

Comment:
Similarly, rename `handleClick` to `onClickPerson` to better describe the action and to follow the naming conventions for event handlers.